src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/FatFunctionPointerNode.cs

```diff
@@ -60,8 +60,9 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
         {
             var builder = new ObjectDataBuilder(factory, relocsOnly);
 
-            // These need to be aligned the same as methods because they show up in same contexts
-            builder.RequireInitialAlignment(factory.Target.MinimumFunctionAlignment);
+            // These need to be aligned the same as method pointers because they show up in same contexts
+            // (macOS ARM64 has even stricter alignment requirement for the linker, so round up to pointer size)
+            builder.RequireInitialAlignment(factory.Target.PointerSize);
```

Comment:
This one is actually hell to diagnose because it produces no error but causes corruption when linking. If absolute relocation of a symbols is 4-byte aligned and spans between two pages in the executable then only the lower part of the relocation is written correctly into the final executable.

This was the root cause of the [intermittent problem](https://github.com/dotnet/runtime/issues/67232#issuecomment-1240624936) I've hit earlier.